File: sdks/go/pulse_test.go
package pulse

import (
	"testing"
)

func TestNewClient(t *testing.T) {
	client := NewClient("http://localhost:8080", Options{
		APIKey: "test-key",
	})

	if client == nil {
		t.Fatal("expected non-nil client")
	}
	if client.url != "http://localhost:8080" {
		t.Errorf("unexpected url: %s", client.url)
	}
	if client.apiKey != "test-key" {
		t.Errorf("unexpected api key: %s", client.apiKey)
	}
}

func TestHealthWhenBrokerDown(t *testing.T) {
	client := NewClient("http://localhost:19999", Options{})
	healthy, err := client.Health()
	if err == nil && healthy {
		t.Error("expected health check to fail when broker is down")
	}
}

File: sdks/typescript/tests/pulse.test.ts
import { Pulse } from '../src/index';

describe('Pulse TypeScript SDK', () => {
  test('can create client', () => {
    const client = new Pulse('http://localhost:8080');
    expect(client).toBeDefined();
  });

  test('can create client with options', () => {
    const client = new Pulse({
      url: 'http://broker:8080',
      apiKey: 'psk_live_abc',
      autoReconnect: false,
    });
    expect(client).toBeDefined();
  });

  test('health check returns false when broker is down', async () => {
    const client = new Pulse('http://localhost:19999');
    const healthy = await client.health();
    expect(healthy).toBe(false);
  });
});
